src/nss-resolve/nss-resolve.c

```diff
@@ -110,6 +110,20 @@ static int count_addresses(sd_bus_message *m, int af, const char **canonical) {
         return c;
 }
 
+static uint32_t ifindex_to_scopeid(int family, const void *a, int ifindex) {
+        struct in6_addr in6;
+
+        if (family != AF_INET6)
+                return 0;
+
+        /* Some apps can't deal with the scope ID attached to non-link-local addresses. Hence, let's suppress that. */
+
+        assert(sizeof(in6) == FAMILY_ADDRESS_SIZE(AF_INET));
+        memcpy(&in6, a, sizeof(struct in6_addr));
```

Comment:
it's a ptr into the bus message, specifically an "ay" byte array, which might have byte-alignment... That's why a copied the data our here, instead of casting...